vaultSDK/lib/udAttributes.dart

```diff
@@ -0,0 +1,730 @@
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+
+import 'udSdkLib.dart';
+
+class UdAttributeSet extends UdSDKClass {
+  Pointer<udAttributeSet> _attributeSet;
+  UdAttributeSet() {
+    this._attributeSet = allocate();
+  }
+
+  /// Creates a udAttributeSet
+  ///
+  /// @param pAttributeSet The attribute set to allocate into
+  /// @param content The standard attributes that will be created, provided as bitfields
+  /// @param additionalCustomAttributes The count of additional attributes to generate, these will be added to the attribute set blank after the standard attributes
+  /// @return A udError value based on the result of the creation of the attribute set.
+  /// @note The application should call udAttributeSet_Free with pAttributeSet to destroy the object once it's no longer needed.
+  ///
+  void create(int udStdContent, int additionalCustomAttributes) {
+    handleUdError(_udAttributeSet_Create(
+        _attributeSet, udStdContent, additionalCustomAttributes));
+  }
+
+  /// Free the memory created by a call to udAttributeSet_Generate
+  ///
+  /// @param pAttributeSet The attribute set to free the resources of
+  /// @return A udError value based on the result of the destruction of the attribute set.
+  void destroy() {
+    handleUdError(_udAttributeSet_Destroy(_attributeSet));
+  }
+
+  /// Gets the offset for a standard attribute so that further querying of that attribute can be performed
+  ///
+  /// @param pAttributeSet The attribute set to get the offset for
+  /// @param attribute The enum value of the attribute
+  /// @param pOffset This pointer will be written to with the value of the offset if it is found
+  /// @return A udError value based on the result of writing the offset to pOffset
+  void getOffsetOfStandardAttribute(int udStdAttribute, int offset) {
+    final Pointer<Uint32> pOffset = allocate();
+    pOffset.value = offset;
+    final err = _udAttributeSet_GetOffsetOfStandardAttribute(
+        _attributeSet, udStdAttribute, pOffset);
+    free(pOffset);
+
+    handleUdError(err);
+  }
+
+  /// Gets the offset for a standard attribute so that further querying of that attribute can be performed
+  ///
+  /// @param pAttributeSet The attribute set to get the offset for
+  /// @param attribute The enum value of the attribute
+  /// @param pOffset This pointer will be written to with the value of the offset if it is found
+  /// @return A udError value based on the result of writing the offset to pOffset
+  void getOffsetOfNamedAttribute(String name, int offset) {
+    final Pointer<Uint32> pOffset = allocate();
```

Comment:
You don't call `free` for this.

Reply:
Are we doing this completely wrong then? I just looked at what the function does and I'm thinking we should either:
- return an `int` from the wrapper function `getOffsetNamedAttribute`
- leave the return type as `void` and set the value a class property

Reply:
My apologies, I didn't look very closely at the function itself, just that it differed to the one above. Yes, this should return an `int` (the offset), so it should look something like:
```dart
int getOffsetOfNamedAttribute(String name) {
  final Pointer<Uint32> pOffset = allocate();
  final pName = Utf8.toUtf8(name);
  handleUdError(_udAttributeSet_GetOffsetOfNamedAttribute(
      _attributeSet, pName, pOffset));
  final offset = pOffset.value;
  free(pName);
  free(pOffset);
  return offset;
}
```

Reply:
Also, to clarify, the function above this one should be adjusted similarly.

Reply:
Updated now 262d68d